EIPS/eip-x.md

```diff
@@ -0,0 +1,140 @@
+| Blocks regularly under-full, user wants urgent inclusion.           | User over-pays, but reliably gets transaction included.                                                                                                         | User bids at or over current price tier, gets transaction mined reliably.                    | User pays an expected amount, and gets transaction mined reliably.                                                 |
+| Blocks regularly under-full, user willing to wait for a good price. | User bids below the low end of the recently accepted prices, may need to wait for a while. If waiting too long, user may need to re-submit with a higher price. | User chooses their price, and waits for it, or manually re-submits.                          | User chooses their lowest price, but also their highest price and maximum wait time, so no resubmission is needed. |
+
+In all cases, the escalator algorithm as I have described is able to perform optimally.
+
+The current gas auction model works well under half-full and less conditions, but for users with urgent needs, has the downside of overpayment. For users seeking a low price, the current model has the downside of requiring re-submission, but has the benefit of always giving users a path towards reliable block inclusion.
+
+EIP-1559 also performs well under normal conditions, but under conditions where blocks are regularly full, the price discovery mechanism breaks, and miners will fall back to the `TIP` parameter to choose the transactions to include, meaning that under network congestion, EIP-1559 forces users to _either_ choose efficient prices or certainty of next-block inclusion.
+
+EIP-1559 also has all the re-submission issues of the current model in situations where a user would like to pay under the current market rate, but has certain time constraints limiting their patience. The Escalator algorithm is the only strategy listed here that allows users to discover the lowest possible price given the network conditions and their time constraints.
+
+## Specification
+<!--The technical specification should describe the syntax and semantics of any new feature. The specification should be detailed enough to allow competing, interoperable implementations for any of the current Ethereum platforms (go-ethereum, parity, cpp-ethereum, ethereumj, ethereumjs, and [others](https://github.com/ethereum/wiki/wiki/Clients)).-->
+
+**Client-Wide Parameters**
+* `INITIAL_FORK_BLKNUM`: TBD
+
+**Transaction Parameters**
+The transaction `gasPrice` parameter is now optional, and if excluded can be replaced by these parameters instead:
+
+* `START_PRICE`: The lowest price that the user would like to pay for the transaction.
+* `START_BLOCK`: The first block that this transaction is valid at.
```

Comment:
How is the `START_BLOCK` actually defined? If a transaction could have been valid at block 1000, but it has been created at block 2000, what is the `START_BLOCK` in such case?

Reply:
Similarly, what happens if `START_BLOCK` is in the future?  Can this EIP *also* solve the problem of delayed transactions?

Reply:
My question was basically a hint that `START_BLOCK` probably cannot be suitably defined (so that it can be efficiently calculated from consensus values, and not just from subjective tx pool timestamps), and this could be the main thing that makes this EIP hard or impossible to implement

Reply:
What do you mean suitably defined? The user or user's wallet specifies the `START_BLOCK` and `MAX_BLOCK` in the transaction, just like with gas price and gas limit. If the miner is working on a block with number `BLOCK_NUMBER`, the block is valid iff _all_ transactions within the block have `START_BLOCK` <= `BLOCK_NUMBER` and `MAX_BLOCK` >= `BLOCK_NUMBER`

It's very easy for a wallet to set this parameter. Just choose the next block for `START_BLOCK` and the `MAX_BLOCK` to whatever delay the user is willing to tolerate divided by the 15 seconds average block times plus the start block.

This constraint can be cheaply validated by any node, and adds some compute time to ordering transactions into a block (must determine the gas price for each transaction for the given block, and whether it can be included) but also has the benefit that transactions can be dropped from the mempool as soon as the block number exceeds `MAX_BLOCK` by some threshold, and potentially if they are sent with a `START_BLOCK` too far in the future

Reply:
Ok, I misunderstood, now I see that `START_BLOCK` is suggested to be a new transaction parameter. Thank you